python/cudf/cudf/_lib/cudf.pyx

```diff
@@ -396,93 +329,82 @@ cdef gdf_column_to_column(gdf_column* c_col, int_col_name=False):
         A flag indicating the string column name should be cast
         to an integer after decoding (default: False).
     """
-    from cudf.core.column import Column
+    from cudf.core.buffer import Buffer
+    from cudf.core.column import build_column, as_column
+    from cudf.utils.utils import mask_bitsize, calc_chunk_size
+
     name = None
-    ncount = c_col.null_count
     if c_col.col_name is not NULL:
         name = c_col.col_name.decode()
         if int_col_name:
             name = int(name)
-    data, mask = gdf_column_to_column_mem(c_col)
-    return Column.from_mem_views(data, mask, ncount, name)
+    
+    gdf_dtype = c_col.dtype
+    data_ptr = int(<uintptr_t>c_col.data)
 
-
-cdef gdf_column_to_column_mem(gdf_column* input_col):
-    gdf_dtype = input_col.dtype
-    data_ptr = int(<uintptr_t>input_col.data)
     if gdf_dtype == GDF_STRING:
         data = nvstrings.bind_cpointer(data_ptr)
+        result = as_column(data, name=name)
     elif gdf_dtype == GDF_STRING_CATEGORY:
         # Need to do this just to make sure it's freed properly
         garbage = rmm.device_array_from_ptr(
             data_ptr,
-            nelem=input_col.size,
+            nelem=c_col.size,
             dtype='int32',
             finalizer=rmm._make_finalizer(data_ptr, 0)
         )
-        if input_col.size == 0:
+        if c_col.size == 0:
             data = nvstrings.to_device([])
         else:
-            nvcat_ptr = int(<uintptr_t>input_col.dtype_info.category)
+            nvcat_ptr = int(<uintptr_t>c_col.dtype_info.category)
             nvcat_obj = nvcategory.bind_cpointer(nvcat_ptr)
             data = nvcat_obj.to_strings()
+        result = as_column(data, name=name)
+        
     else:
-        data = rmm.device_array_from_ptr(
-            data_ptr,
-            nelem=input_col.size,
-            dtype=np_dtype_from_gdf_column(input_col),
-            finalizer=rmm._make_finalizer(data_ptr, 0)
-        )
-
-    mask = None
-    if input_col.valid:
-        mask_ptr = int(<uintptr_t>input_col.valid)
-        mask = rmm.device_array_from_ptr(
-            mask_ptr,
-            nelem=calc_chunk_size(input_col.size, mask_bitsize),
-            dtype=mask_dtype,
-            finalizer=rmm._make_finalizer(mask_ptr, 0)
+        dtype = np_dtype_from_gdf_column(c_col)
+        dbuf = rmm.DeviceBuffer(
+            ptr=data_ptr,
+            size=dtype.itemsize * c_col.size,
         )
+        data = Buffer(dbuf)
+        mask = None
+        if c_col.valid:
```

Comment:
ambiguous
```suggestion
        if c_col.valid is not None:
```

Reply:
`c_col.valid` is a `bool` I think.

Reply:
Triaged and `c_col` is a `gdf_column` and it's `valid` member is a void pointer to the validity mask, so just check to make sure it's not nullptr